fvm/evm/impl/impl.go

```diff
@@ -648,21 +649,30 @@ func newInternalEVMTypeWithdrawFunction(
 				panic(errors.NewUnreachableError())
 			}
 
-			amount := types.NewBalance(amountValue.BigInt)
+			_, overflow := uint256.FromBig(amountValue.BigInt)
+			if overflow {
+				panic(types.ErrInvalidBalance)
+			}
+
+			if types.BalanceInAttFlowValidForFlowVault(amountValue.BigInt) {
```

Comment:
Based on the name, I would expect true here to be valid. But it seems the check results in a `panic` so I'm probably misunderstanding the meaning of `BalanceInAttFlowValidForFlowVault`. What is being checked here?

Reply:
@sisyphusSmiling My bad :sweat_smile: The name and the logic didn't match the intention.

I have updated it in https://github.com/onflow/flow-go/pull/6877/commits/f0e23ff955917079a31e08d6127d1440e599f6d1.

This method basically checks if the requested withdrawal amount in attoFlow, can be represented in a FlowToken vault. The smallest allowed withdrawal amount is `1e^10` attoFlow, which is `1e^-8` Flow (or 0.00000001).

I have chosen to panic on such a case, as there's no reason to silently allow such cases.